sdk/storage/Microsoft.Azure.Management.Storage/CHANGELOG.md

```diff
@@ -1,5 +1,20 @@
 ## Microsoft.Azure.Management.Storage release notes
 
+### Changes in 18.0.0-beta
+- Upgrade to rest api version 2020-08-01-preview
+- Support enanble ContainerDeleteRetentionPolicy in BlobServices properties
+- Support enable LastAccessTimeTrackingPolicy in BlobServices properties
+- Support DaysAfterLastAccessTimeGreaterThan,EnableAutoTierToHotFromCool in ManagementPolicy BaseBlob Actions
+- Support List deleted containers
+- Support enable ProtocolSettings.Smb.Multichannel on FileServices properties
+- Support ResourceAccessRule in Microsoft.Azure.Management.Storage.Models.NetworkRuleSet
+- Support Set ExtendedLocation in create storage account
+- Support Blob Inventory rule of storage account
+
+**Breaking changes**
+
+- In StorageManagementClient.FileServices.SetServiceProperties(), add a madatory parameter with type Microsoft.Azure.Management.Storage.Models.FileServiceProperties, to input all FileService properties, and remove 2 parameters to input FileService properties: cors， shareDeleteRetentionPolicy.
```

Comment:
Is this just a temporary change for the preview, or do you intend to go through with this change in the stable API?  This will need to go through a breaking change review before you can release as a stable api

Reply:
This breaking will also in SDK of stable API version. How to go through a breaking change review？
I would suggest to merge this PR then start the process to breaking review (or resolve the breaking).

Actually, this is not caused by the swagger (or API) breaking, but caused by a limitation of autorest.

We have config `payload-flattening-threshold: 2` in https://github.com/Azure/azure-rest-api-specs/blob/master/specification/storage/resource-manager/readme.csharp.md#common-c-settings, so when a object has <= 2 child properties, in autorest generated code, functions will use the child properties as input in stead of the object; when the object has >=3 properties,  functions will  use the object as input. 

So when `FileServiceProperties ` properties count increase from 2 to 3, the function `SetServiceProperties() ` will change from input the file service properties to input FileServiceProperties object.

This kind of breaking happens frequently in .net SDK. Do you have any idea how to resolve this?
If we change the `payload-flattening-threshold: 2` to a bigger value, it will also cause many of the existing API breaking.

Reply:
We can likely solve it in individual cases with customizations.  Can you send me an email, so I can bring in some generator folks?